### PR TITLE
trying to porting to chisel3 sytle, be easy to use in other projects.

### PR DESCRIPTION
**Pull request by sequencer**
**Related issue**: 
https://github.com/ucb-bar/asyncqueue/pull/3
https://github.com/freechipsproject/chisel3/issues/1067

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: proposal

**Release Notes**
No need to merge these commit, just PR for test, to check if these code is ok in rocketchip.
If possible, will utils be progressively merged into chisel3.experimantal?
these utils could benefit chisel3 standard library well.

### CODE REVIEW

File: src/main/scala/util/ShiftReg.scala [hunk 1]
```diff
@@ -2,107 +2,115 @@
...
     sink.io.d := source.io.q
-    sink.io.en := Bool(true)
+    sink.io.en := true.B
   }
   io.q := chain.head.io.q
 }
 
 object AsyncResetShiftReg {
-  def apply [T <: Chisel.Data](in: T, depth: Int, init: Int  = 0, name: Option[String] = None): T =
+  def apply[T <: Chisel.Data](in: T, depth: Int, init: Int = 0, name: Option[String] = None): T =
```
Comment (edwardcwang) [inline src/main/scala/util/ShiftReg.scala]: FYI in Chisel3 this can just be `T <: Data` as opposed to `Chisel.Data`

**Comment by edwardcwang** [comment]
@sequencer ```
[error] /home/travis/build/freechipsproject/rocket-chip/src/main/scala/util/AsyncResetReg.scala:46:3: not found: value addResource
[error]   addResource("/vsrc/AsyncResetReg.v")
[error]   ^
```

**Comment by sequencer** [comment]
So sorry I forgot this PR. 
Just Increased priority on this. ;)

**Comment by sequencer** [comment]
I just update all util to chisel3, just for test.

**Pull request closed without merging**